src/xserver.cpp

```diff
@@ -77,8 +77,7 @@ namespace xeus
         m_stdin_listener(message);
     }
 
-    std::unique_ptr<xserver> make_xserver(zmq::context_t& context,
-                                          const xconfiguration& config)
+    std::unique_ptr<xserver> make_xserver(zmq::context_t& context, xconfiguration& config)
```

Comment:
The `config` parameter should remain constant. A method that fill configuration parameter mus be added to the `xserver` API so the kernel can update its configuration after the server is built if the port numbers were not filled in the original configuration.